kentikapi/client.go

```diff
@@ -174,58 +203,38 @@ func makeRetryingClientConfig(c Config) httputil.ClientConfig {
 	}
 }
 
-//nolint: gosec
-func makeConnForSynthetics(c Config) grpc.ClientConnInterface {
-	if c.SyntheticsGRPCAPIURL == "" {
-		c.SyntheticsGRPCAPIURL = syntheticsGRPCAPIURL
-		config := &tls.Config{}
-		conn, err := grpc.Dial(
-			syntheticsGRPCAPIURL,
-			grpc.WithTransportCredentials(credentials.NewTLS(config)),
-			grpc.WithUnaryInterceptor(c.createUnaryInterceptor()),
-		)
-		if err != nil {
-			log.Printf("cannot connect to api url for synthetics: %v", err)
-		}
-		return conn
-	}
-	conn, err := grpc.Dial(c.SyntheticsGRPCAPIURL, grpc.WithInsecure())
-	if err != nil {
-		log.Printf("cannot connect to api url for synthetics: %v", err)
-	}
-	return conn
-}
-
-//nolint: gosec
-func makeConnForCloudExport(c Config) grpc.ClientConnInterface {
-	if c.CloudExportGRPCAPIURL == "" {
-		c.CloudExportGRPCAPIURL = cloudExportGRPCAPIURL
-		config := &tls.Config{}
+func (c Config) makeConnForGRPC(hostPort string) (grpc.ClientConnInterface, error) {
+	if !c.DisableTLS {
+		config := &tls.Config{MinVersion: tls.VersionTLS13}
 		conn, err := grpc.Dial(
```

Comment:
I still think that there could be one grpc.Dial() call in this function. This if statement block could only set tlsInterceptor local variable, that can be used in grpc.Dial() below, eg:
```go
func (c Config) makeConnForGRPC(hostPort string) (grpc.ClientConnInterface, error) {
	return grpc.Dial(
		hostPort,
		c.tlsOption(),
		grpc.WithUnaryInterceptor(c.createUnaryInterceptor()),
	)
}

func (c Config) tlsOption() grpc.DialOption {
	if c.DisableTLS {
		return grpc.WithInsecure()
	}
	return grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{
		MinVersion: tls.VersionTLS13,
	}))
}
```

Note that tlsOption could also be `makeTLSOption`.

Also, we have `grpc synthetics connection: %v` error string above, I think that `cannot connect to api` can be omitted, because it basically duplicates information.

Reply:
Done